source/MaterialXGenShader/ShaderGraph.cpp

```diff
@@ -490,6 +487,12 @@ ShaderGraphPtr ShaderGraph::createSurfaceShader(
                 inputSocket->setUnit(nodeInputUnit);
                 input->setUnit(nodeInputUnit);
             }
+            const string& nodeColorspace = nodeInput->getColorSpace();
+            if (!nodeColorspace.empty())
+            {
+                inputSocket->setColorspace(nodeColorspace);
+                input->setColorspace(inputSocket->getColorspace());
```

Comment:
Nitpick: Why not assign nodeColorspace here as well

Reply:
Makes sense. Cut and pasted. Will fix the other place also.